src/Repository/AbstractRepository.php

```diff
@@ -184,10 +219,16 @@ public function getEntityName(): string
      */
     public function getTableName(): string
     {
-        //ReallyOrm\Test\Entity\User
-        preg_match('/.*\\\\(.*)/', $this->entityName, $matches);
+        //ReallyOrm\Test\Entity\UserTemplate
+        preg_match('/.*\\\\(.*)/', $this->entityName, $matches); // UserTemplate
+        $tableName = lcfirst($matches[1]); // userTemplate
+        if (ctype_lower($tableName)) {
+            return $tableName;
+        }
+        $pieces = preg_split('/(?=[A-Z])/', $tableName); // 0 => user; 1 => Template
+        $tableName = $pieces[0] . '_' . lcfirst($pieces[1]); // user_template
```

Comment:
impressive work, you can also use stackoverflow when in need :)) 
https://stackoverflow.com/questions/1993721/how-to-convert-pascalcase-to-pascal-case

Reply:
also, there is always the option to pass the repository the table name and not go through all this processing. if the repo knows about the entity name, why can't it know about the table name as well? (and you would also get rid of this convention that entity is TitleCase equivalent of table which is snake_case)